### PR TITLE
parallel netcdf writes, inpost with no thread 

### PR DESCRIPTION
**Pull request by junwang-noaa**
This pull request has following code changes

1)fv3atm pull request 41: parallel netcdf IO. fv3tam issue 23 parallel netcdf writes.
2) update post lib to not compile with "openmp". fv3atm issue 49:inline post does not reproduce with threads.

FV3ATM pull request 41: https://github.com/NOAA-EMC/fv3atm/pull/41

#43:

### CODE REVIEW

**Comment by jswhit2** [comment]
The files in the cmake directory need to have the PARALLEL_NETCDF option added

File: conf/configure.fv3.wcoss_cray [hunk 1]
```diff
@@ -47,7 +47,7 @@ CFLAGS := $(INCLUDE)
 FFLAGS := $(INCLUDE) -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -nowarn -sox -align array64byte
 
 CPPDEFS += -Duse_libMPI -Duse_netCDF -DSPMD -DUSE_LOG_DIAG_FIELD_INFO  -DUSE_GFSL63 -DGFS_PHYS -Duse_WRTCOMP
-CPPDEFS += -DNEW_TAUCTMAX -DINTERNAL_FILE_NML
+CPPDEFS += -DNEW_TAUCTMAX -DINTERNAL_FILE_NML -DNO_PARALLEL_NETCDF
```
Comment (yangfanglin) [inline conf/configure.fv3.wcoss_cray:50]: Parallel netCDF not supported on Cray ?  It is fine  if it is the case.  The global workflow needs to have different I/O settings on different platforms.

**Comment by junwang-noaa** [comment]
I got some error when installing the hdf1.10.6 on cray. I am asking
George's help to see which cc compiler I should use on cray. So at this
time I disabled the parallel netcdf on cray.

**Review by yangfanglin — APPROVED** [review]
Read through the changes. I think they are all fine. Knowing the code helps me to think how to generalize the workflow scripts to run the model for various options and on different platforms.

**Comment by jedwards4b** [comment]
@junwang-noaa Why do you need to install parallel-netcdf on cray?  It is one of their standard modules, if it's not already on the system, you should be able to get support staff to install it.  
Cray supported netcdf and parallel-netcdf modules are: 
cray-parallel-netcdf
and cray-netcdf

**Comment by junwang-noaa** [comment]
This is for the new netcdf lib with parallel writing compressed data. We
cant ask system adm to install a non public release version on operational
machine. we install it ourselves  for the testing purpose.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**